Added empty state in card to empty state guidelines

mmenestr opened the pull request:
@mcarrano let me know if what I added looks good to you. It's at the very bottom of the guidelines, and I tried to address the feedback we got at the PF share!

patternfly-build commented:
PF3 preview: https://patternfly-org-pr-2294-v3.surge.sh/v3

mcarrano commented on packages/v4/src/content/design-guidelines/components/data-list/data-list.md:
```diff
@@ -53,27 +53,29 @@ Every data list can be extended with these functionalities:
 
-1. **Expansion caret:** Toggles the expansion open and closed.
-2. **Expansion panel:** A container that is revealed when the expansion is open. It can accept any supported layout to present additional content to the user.
+1. **Expand all:** Expands or collapses every row in the data list at the same time.
```
Should this be marked as 'optional'? I don't think you would always require this feature.

mcarrano commented on packages/v4/src/content/design-guidelines/components/data-list/data-list.md:
```diff
@@ -53,27 +53,29 @@ Every data list can be extended with these functionalities:
-2. **Checkbox:** Enables a user to select a row.
-3. **Global actions:** Actions that can be applied to all selected items.
+2. **Checkbox:** Enables a user to select a row. Use when multiple rows can be selected at the same time.
+3. **Radio button:** Enables a user to select a row. Use when only one row can be selected at a time.
```
Even though this seems perfectly valid, we didn't actually implement this feature for data lists.

mmenestr replied:
Removed it from data list then!

mcarrano commented on packages/v4/src/content/design-guidelines/components/empty-state/empty-state.md:
```diff
@@ -189,6 +189,10 @@ In some situations, users may need to add or create something to view associated
 
 <img src="./img/add-or-create.png" alt="Example of add empty state" width="990"/> 
 
+You may also use an extra-small empty state inside of a card to present the user with another meands of adding a new card/item to their card view. If used, we recommend using it **in addition to** a primary button in a toolbar, to ensure that the user is still able to perform the action no matter what page of the card view they are on. We also recommend placing the addition action card as the ver ...
```
Maybe add that placement should be consistent with where a new card will be added.

mcarrano commented on packages/v4/src/content/design-guidelines/components/table/table.md:
```diff
@@ -47,10 +47,11 @@ Every table can be extended with these functionalities:
 
-1. **Expansion:** Expands the row
-2. **Expansion panel:** Contains details associated with a row. See [expanded panel coloring](#expanded-panel-coloring) for more information.
+1. **Expand all**: Expands every row at the same time.
```
Should be optional?

mcarrano reviewed (changes requested):
@mmenestr Looks good. See my comments. I'm also wondering if we should add something about using the empty state card in the Card View documentation here: https://patternfly-org-pr-2294-v4.surge.sh/v4/components/card/design-guidelines#card-view-usage That's where I would have expected to look for it.

mmenestr commented:
@mcarrano please review again. I added the empty state card to the card view guidelines, and removed the radio button mention from the data list! + added "optional" for the expand all!

mcarrano reviewed (approved):
Looks good @mmenestr . Thanks for making these changes. I will merge.

patternfly-build commented:
Your changes have been released in: 
 - patternfly-org-4@4.2.87

Thanks for your contribution! :tada:
